src/inbox/mod_inbox_muc.erl

```diff
@@ -92,28 +95,28 @@ direction(From, To) ->
       To :: receiver_bare_user_jid(),
       Packet :: packet().
 handle_outgoing_message(Host, Room, To, Packet) ->
-    Markers = mod_inbox_utils:get_reset_markers(Host),
-    case mod_inbox_utils:if_chat_marker_get_id(Packet, Markers) of
-        undefined ->
-            mod_inbox_utils:write_to_sender_inbox(Host, To, Room, Packet);
-        Id ->
-            mod_inbox_utils:reset_unread_count(To, Room, Id)
-    end.
+    maybe_reset_unread_count(Host, To, Room, Packet),
+    maybe_write_to_inbox(Host, To, Room, Packet, fun write_to_sender_inbox/4).
 
 -spec handle_incoming_message(Host, Room, To, Packet) -> term() when
       Host :: receiver_host(),
       Room :: room_bare_jid(),
       To :: receiver_bare_user_jid(),
       Packet :: packet().
 handle_incoming_message(Host, Room, To, Packet) ->
-    Markers = mod_inbox_utils:get_reset_markers(Host),
-    case mod_inbox_utils:has_chat_marker(Packet, Markers) of
-        true ->
-            %% don't store chat markers in inbox
-            ok;
-        false ->
-            mod_inbox_utils:write_to_receiver_inbox(Host, Room, To, Packet)
-    end.
+    maybe_write_to_inbox(Host, Room, To, Packet, fun write_to_receiver_inbox/4).
+
+maybe_reset_unread_count(Host, User, Room, Packet) ->
```

Comment:
Any real benefit from extracting these 4 functions? Formatting? Could be solved if `-import` as well.

Reply:
More granularity when mocking/tracing.